api/app/Extensions/JWTAuth/JWTAuth.php

```diff
@@ -0,0 +1,58 @@
+<?php namespace App\Extensions\JWTAuth;
+
+use Illuminate\Http\Request;
+use Tymon\JWTAuth\Token;
+use Tymon\JWTAuth\JWTAuth as JWTAuthBase;
+use RuntimeException;
+use App\Exceptions\BadRequestException;
+use App\Exceptions\UnauthorizedException;
+use App\Exceptions\TokenInvalidException;
+use App\Exceptions\UnprocessableEntityException;
+use Tymon\JWTAuth\Exceptions\TokenExpiredException;
+
+class JWTAuth extends JWTAuthBase
+{
+    /**
+     * Get the token contained within the current request.
+     *
+     * @param  Request  $request
+     * @throws BadRequestException
+     * @return Token
+     */
+    public function getTokenFromRequest(Request $request)
```

Comment:
The request object being injected in the constructor is not yet treated by the framework, and is lacking the header information, thus injecting the request with setrequest() takes care of that. I'll see if we can change to have the completed request object injected in the constructor instead.